dcmgr/lib/dcmgr/models/dhcp_range.rb

```diff
@@ -15,6 +15,33 @@ class DhcpRange < BaseNew
       new_dataset
     }
 
+    # Query ranges which exist between begin_u32 and end_u32.
+    # Assume: 192.168.0.0/24: 10-20, 30-40, 50-60
+    # included_ranges("192.168.0.30", "192.168.0.70")
+    #   => 192.168.0.30-40, 192.168.0.50-60
+    def_dataset_method(:included_ranges) {|begin_u32, end_u32|
+      self.filter("range_begin >= ? AND range_end <= ?", begin_u32, end_u32)
+    }
+
+    # Query ranges which contain the ipv4_u32 address(es).
+    # Assume: 192.168.0.0/24: 10-20, 30-40, 50-60
+    # hit_ranges("192.168.0.10", "192.168.0.55")
+    #   => 192.168.0.10-20, 192.168.0.50-60
+    def_dataset_method(:hit_ranges) { |*ipv4_u32|
+      sql = "(range_begin <= ? AND range_end >= ?)"
+      ary = ipv4_u32.map { |u32|
+        [sql, [u32, u32]]
+      }
+      sql = "(" + ary.map { |sql, args|
+        sql
+      }.join(") OR (") + ")"
+      filter("(" + sql + ")",
+             *ary.map {|sql, args|
+               args
+             }.flatten
+             ).order(Sequel.asc(:range_begin))
+    }
```

Comment:
This code is very hard to read. It could use some newlines.

```ruby
      sql = "(range_begin <= ? AND range_end >= ?)"

      ary = ipv4_u32.map { |u32|
        [sql, [u32, u32]]
      }

      sql = "(" + ary.map { |sql, args|
        sql
      }.join(") OR (") + ")"

      filter("(" + sql + ")",
             *ary.map {|sql, args|
               args
             }.flatten
             ).order(Sequel.asc(:range_begin))

```